src/saltext/vmware/states/vmc_distributed_firewall_rules.py

```diff
@@ -0,0 +1,487 @@
+        (Optional) Flag to disable the rule. Default is false.
+
+    logged
+        (Optional) Enable logging flag. Flag to enable packet logging. Default is disabled.
+
+    description
+        (Optional) Description of this resource
+
+    direction
+        (Optional) Define direction of traffic
+        Possible Values for are: IN, OUT, IN_OUT
+        Default: "IN_OUT"
+
+    notes
+        (Optional) Text for additional notes on changes.
+
+    tag
+        (Optional) Tag applied on the rule. User level field which will be printed in CLI and packet logs.
+
+    tags
+        (Optional) Opaque identifiers meaningful to the user.
+
+        .. code-block::
+
+            {
+                "tag": <tag>,
+                "scope": <scope>
+            }
+
+    Example values:
+
+        .. code-block::
+
+           {
+              "description": "comm entry",
+              "display_name": "",
+              "sequence_number": 1,
+              "source_groups": [
+                "ANY"
+              ],
+              "logged": false,
+              "destination_groups": [
+                "ANY"
+              ],
+              "scope": [
+                "ANY"
+              ],
+              "action": "DROP",
+              "services": [
+                "ANY"
+              ],
+              "direction": "IN_OUT",
+              "tag": "",
+              "notes": ""
+            }
+
+    """
+
+    rule_id = name
+    input_dict = {
+        "source_groups": source_groups,
+        "destination_groups": destination_groups,
+        "services": services,
+        "scope": scope,
+        "action": action,
+        "display_name": display_name,
+        "sequence_number": sequence_number,
+        "disabled": disabled,
+        "logged": logged,
+        "description": description,
+        "direction": direction,
+        "notes": notes,
+        "tag": tag,
+        "tags": tags,
+    }
+
+    input_dict = {k: v for k, v in input_dict.items() if v != vmc_constants.VMC_NONE}
+
+    distributed_firewall_rule = __salt__["vmc_distributed_firewall_rules.get_by_id"](
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        domain_id=domain_id,
+        security_policy_id=security_policy_id,
+        rule_id=rule_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+    if "error" in distributed_firewall_rule:
+        if "could not be found" in distributed_firewall_rule["error"]:
+            distributed_firewall_rule = None
+        else:
+            return vmc_state._create_state_response(
+                name=name, comment=distributed_firewall_rule["error"], result=False
+            )
+
+    if __opts__.get("test"):
+        log.info("present is called with test option")
+        return vmc_state._create_state_response(
+            name=name,
+            comment="Distributed firewall rule {} will be {}".format(
```

Comment:
**suggestion** it's more common in salt test modes to say `"would have been"` (because nothing is happening nor will it happen unless they actually run the state in non-test mode)

Reply:
updated to `would have been`